rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -311,93 +311,36 @@ void getPendingFederation_nullPendingFederationInStorage_returnsNull() {
-            createFederation(NON_STANDARD_ERP_FEDERATION_FORMAT_VERSION)
-        );
-
-        testSaveNewFederation(
-            P2SH_ERP_FEDERATION_FORMAT_VERSION,
-            createFederation(P2SH_ERP_FEDERATION_FORMAT_VERSION)
-        );
-    }
-
-    @Test
-    void saveNewFederation_after_RSKIP123_should_save_standard_multisig_fed_format() {
-        activations = ActivationConfigsForTest.only(ConsensusRule.RSKIP123).forBlock(0);
-        testSaveNewFederation(
-            STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION,
-            createFederation(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION)
-        );
-    }
-
-    @Test
-    void saveNewFederation_after_RSKIP201_should_save_standard_multisig_fed_format() {
-        activations = ActivationConfigsForTest.only(
-            ConsensusRule.RSKIP123,
-            ConsensusRule.RSKIP201
-        ).forBlock(0);
-        testSaveNewFederation(
-            STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION,
-            createFederation(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION)
-        );
-    }
-
-    @Test
-    void saveNewFederation_after_RSKIP353_should_save_standard_multisig_fed_format() {
-        activations = ActivationConfigsForTest.only(
-            ConsensusRule.RSKIP123,
-            ConsensusRule.RSKIP201,
-            ConsensusRule.RSKIP353
-        ).forBlock(0);
-        testSaveNewFederation(
-            STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION,
-            createFederation(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION)
-        );
-    }
-
-    @Test
-    void saveNewFederation_after_RSKIP201_should_save_non_standard_erp_fed_format() {
-        activations = ActivationConfigsForTest.only(
-            ConsensusRule.RSKIP123,
-            ConsensusRule.RSKIP201
-        ).forBlock(0);
-        testSaveNewFederation(
-            NON_STANDARD_ERP_FEDERATION_FORMAT_VERSION,
-            createFederation(NON_STANDARD_ERP_FEDERATION_FORMAT_VERSION)
-        );
-    }
-
-    @Test
-    void saveNewFederation_after_RSKIP353_should_save_non_standard_erp_fed_format() {
-        activations = ActivationConfigsForTest.only(
-            ConsensusRule.RSKIP123,
-            ConsensusRule.RSKIP201,
-            ConsensusRule.RSKIP353
-        ).forBlock(0);
-        testSaveNewFederation(
-            NON_STANDARD_ERP_FEDERATION_FORMAT_VERSION,
-            createFederation(NON_STANDARD_ERP_FEDERATION_FORMAT_VERSION)
-        );
-    }
+    private static Stream<Arguments> provideSaveNewFederationTestArguments() {
 
-    @Test
-    void saveNewFederation_after_RSKIP353_should_save_p2sh_erp_fed_format() {
-        activations = ActivationConfigsForTest.only(
-            ConsensusRule.RSKIP123,
-            ConsensusRule.RSKIP201,
-            ConsensusRule.RSKIP353
-        ).forBlock(0);
-        testSaveNewFederation(
-            P2SH_ERP_FEDERATION_FORMAT_VERSION,
-            createFederation(P2SH_ERP_FEDERATION_FORMAT_VERSION)
+        return Stream.of(
```

Comment:
I don't think so.
During unit tests, there's no need to reuse the same objects. Sending the same `federation` object to different test is not a good idea. Each test should have their instances as new and clean as possible.

Creating a variable for each of these would rather complicate the `provideSaveNewFederationTestArguments` method. We could discuss it further.

Reply:
We can get a cleaner code putting these values apart. The idea is to get the same standards of quality as production code. If you are debugging the code is more complicated to know the value we get with those methods since you can't see what are the input values that are being sent. 

Besides that, we are adding more load to the performance of the tests executing the `createFederation` method with the same inputs several times. It would be great to get fast tests.